apps/src/templates/teacherDashboard/StatsTableWithData.jsx

```diff
@@ -0,0 +1,56 @@
+import React, {Component, PropTypes} from 'react';
+import Spinner from '@cdo/apps/code-studio/pd/components/spinner';
+import StatsTable from './StatsTable';
+
+class StatsTableWithData extends Component {
+  static propTypes = {
+    sectionId: PropTypes.string,
+  };
+
+  state = {
+    studentsCompletedLevelCount: {},
+    section: {},
+    isLoading: true
+  };
+
+  componentDidMount() {
+    const completedLevelsUrl = `/dashboardapi/sections/${this.props.sectionId}/students/completed_levels_count`;
+    $.ajax({
+      url: completedLevelsUrl,
+      method: 'GET',
+      dataType: 'json'
+    }).done(studentsCompletedLevelCount => {
+      this.setState({studentsCompletedLevelCount: studentsCompletedLevelCount});
+    });
+
+    const sectionUrl = `/api/v1/sections/${this.props.sectionId}`;
+    $.ajax({
+      url: sectionUrl,
+      method: 'GET',
+      dataType: 'json'
+    }).done(section => {
+      this.setState({
+        section: section,
+        isLoading: false
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        {this.state.isLoading &&
+          <Spinner/>
+        }
+        {!this.state.isLoading &&
+          <StatsTable
```

Comment:
once we're ready to swap out the old tabs for these, we should delete the `<StatsTable/>` component and just have everything for that tab in one component